Diz.Core/model/byteSources/ByteStorage.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Xml.Serialization;
+
+// TODO: we can probably simplify most of this by replacing the parent class with ParentAwareCollection<ByteSource, ByteEntry>
```

Comment:
TODO: we can probably simplify most of this by replacing the parent class with ParentAwareCollection<ByteSource, ByteEntry>